core/nuxeo-resource.js

```diff
@@ -311,6 +311,9 @@ import './nuxeo-connection.js';
         .then((response) =>
           response.text().then((text) => {
             try {
+              if (this.headers.accept.includes('application/xml')) {
+                return text;
+              }
```

Comment:
Since we have a catch that throws an 'Invalid json' error, regardless of what the accept header is, I think I'd rather switch things around and to the current behavior if we know we are accepting json, something like:
```javascript
if (this.headers.accept.includes('application/json')) {
  try {
    return text ? JSON.parse(text) : {};
  } catch (e) {
    return { error: 'Invalid json' };
  }
} else {
  return text;
}
```
WDYT?

Reply:
I agree we're better off parsing json when needed than matching the XML content type specifically. However we should match the response content type and not the request headers

Reply:
Agree with both 😅